### PR TITLE
feat(tutorial library): use Tag component

### PR DESCRIPTION
**Pull request by BRKalow**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

Integrate the HDS `Tag` component to represent active filters.

Screen Shot 2022-09-07 at 4 49 43 PM

## 🧪 Testing

- [ ] visit https://dev-portal-git-brkfeat-integrate-tags-hashicorp.vercel.app/tutorials/library?product=boundary%2Cpacker
- [ ] Validate that the `Tag` component is used for the selected filters
- [ ] Click one of the `x` buttons
- [ ] Validate the filter is removed

### CODE REVIEW

File: src/components/tag/index.tsx [hunk 1]
```diff
@@ -2,10 +2,14 @@ import { IconX16 } from '@hashicorp/flight-icons/svg-react/x-16'
...
 
-const Tag = ({ text, onRemove }: TagProps) => {
+const Tag = ({ text, onRemove, buttonAriaLabel }: TagProps) => {
 	return (
 		<span className={s.root}>
-			<button className={s.removeButton} onClick={onRemove}>
+			<button
+				className={s.removeButton}
+				onClick={onRemove}
+				aria-label={buttonAriaLabel ?? `Dismiss ${text}`}
```
Comment (BRKalow) [inline src/components/tag/index.tsx:11]: based on the HDS implementation, the `aria-label` appears to default to `Dismiss ${text}`. I also added the `buttonAriaLabel` prop to provide additional flexibility here.

File: src/components/tag/types.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ type NativeButtonProps = JSX.IntrinsicElements['button']
 interface TagProps {
 	onRemove?: NativeButtonProps['onClick']
 	text: string
+	buttonAriaLabel?: string
```
Comment (ashleemboyer) [inline src/components/tag/types.ts:6]: suggestion: rename to `removeButtonAriaLabel` to be more explicit

File: src/views/tutorial-library/components/current-filters/index.tsx [hunk 3]
```diff
@@ -62,13 +62,11 @@ export function CurrentFilters() {
...
 								<li key={labelText}>
-									<button
-										onClick={() => item.refine(refinement)}
-										className={currentFiltersStyle.filterButton}
-										aria-label={`Remove filter ${labelText}`}
-									>
-										{labelText} <IconX16 />
-									</button>
+									<Tag
+										buttonAriaLabel={`Remove filter ${labelText}`}
```
Comment (ashleemboyer) [inline src/views/tutorial-library/components/current-filters/index.tsx:66]: Can you add a TODO above this to test with voice control as a breadcrumb for later? This gets into [Success Criterion 2.5.3 Label in Name](https://www.w3.org/TR/WCAG21/#label-in-name) territory.

File: src/components/tag/index.tsx [hunk 4]
```diff
@@ -2,10 +2,14 @@ import { IconX16 } from '@hashicorp/flight-icons/svg-react/x-16'
 import { TagProps } from './types'
 import s from './tag.module.css'
 
-const Tag = ({ text, onRemove }: TagProps) => {
+const Tag = ({ text, onRemove, removeButtonAriaLabel }: TagProps) => {
```
Comment (ashleemboyer) [inline src/components/tag/index.tsx:5]: suggestion: we should be able to set the default value of `removeButtonAriaLabel` even when it uses another prop:

```tsx
const Tag = ({ text, onRemove, removeButtonAriaLabel = `Dismiss ${text}` }: TagProps) => {
  // ...
}
```

**Review by ashleemboyer — APPROVED** [review]

File: src/views/tutorial-library/components/current-filters/index.tsx (hunk 3 above)
Comment (BRKalow) [reply src/views/tutorial-library/components/current-filters/index.tsx]: Sure thing 👍

**Pull request merged**